vertical-pod-autoscaler/enhancements/4902-delete-oom-pods/README.md

```diff
@@ -0,0 +1,79 @@
+# KEP-4902: Delete OOM Pods
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+   - [Goals](#goals)
+   - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+   - [Test Plan](#test-plan)
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+   - [Update the eviction API](#update-the-eviction-api)
+<!-- /toc -->
+
+## Summary
+
+The default behaviour of VPA Updater is to evict Pods when new resource
+recommendations are available. This works fine for most cases.
+A problem that can arise is when there are multiple replicas and a
+PodDisruptionBudget (PDB) which allows one disruption. Now the eviction is at
+risk, because as soon as multiple replicas run into for example the memory
+limit and get killed they will never recover as the eviciton API will not allow
+any further disruptions.
+
+This proposal addresses the problem by allowing users to enable the deletion of
+pods as a backup if the eviction fails.
+
+## Motivation
+
+The motivation behind the change is to give VPA users a way to recover a
+failing deployment, if updated/increased limits would solve the problem.
+
+### Goals
+
+- Main: allow cluster administrators to enable deletion of pods
+- Secondary: configurable deletion threshold to tune the deletion behaviour
+
+### Non-Goals
+
+- Get rid of or work around the existing eviction behaviour
+
+## Proposal
+
+The proposal is to add `--delete-on-eviction-error` to the VPA to enable
```

Comment:
Please describe behavior of the flag in some more detail. Something like:

- when the flag is true and
- pod eviction fails and
- pod has at least one pod which:
  - is waiting with reason `CrashLoopBackOff` and
  - its last reason for termination is `OOMKilled` and 
  - its `RestartCount` is greater than ${threshold}

VPA will delete the pod.

If I get this right then please clarify why one pod meeting those conditions is enough to trigger eviction?

Also I'm not sure why you want to set a threshold on RestartCount? Could you explain to me why I might want to use value >1?

Reply:
The idea is to only use deletion as a very last resort and still mostly rely on eviction to fix problems. I still wanted to honor PDBs as much as possible, so that deletion is only used when there is no way in sight where eviciton would work.
But the more I think about the threshold flag the more unlikely the scenarios get that I initially thought about, because the other  checks will most likely be the first to not succeed.